ocsp.go

```diff
@@ -407,101 +515,212 @@ func verifyPeerCertificate(callback func(*sync.WaitGroup, []*x509.Certificate) [
 			verifiedChains[i] = append(verifiedChains[i], rca)
 			n++
 		}
-		wg.Add(n)
-		results := callback(&wg, verifiedChains[i])
-		wg.Wait()
+		results := getAllRevocationStatus(verifiedChains[i])
 		for _, r := range results {
-			if r.err != nil {
+			if r.err != nil && (r.code == ocspRevoked || ocspFailOpen == ocspFailOpenFalse) {
```

Comment:
OCSP Fail open mode check.
Except Revoked status, all other errors will be ignored when ocspFailOpen==ocspFailOpenTrue